j5/backends/console/sr/v4/power_board.py

```diff
@@ -0,0 +1,146 @@
+from j5.boards import Board
+from j5.boards.sr.v4.power_board import PowerBoard, PowerOutputPosition
+from j5.components import (
+    BatterySensorInterface,
+    ButtonInterface,
+    LEDInterface,
+    PiezoInterface,
+    PowerOutputInterface,
+)
+
+
+class SRV4PowerBoardConsoleBackend(
+    PowerOutputInterface,
+    PiezoInterface,
+    ButtonInterface,
+    BatterySensorInterface,
+    LEDInterface,
+    Backend,
+):
+    """The hardware implementation of the SR V4 power board."""
+
+    environment = ConsoleEnvironment
+    board = PowerBoard
+
+    @classmethod
+    def discover(cls) -> List[Board]:
+        """Discover boards that this backend can control."""
+        raise NotImplementedError("The Console Backend cannot discover boards.")
+
+    def __init__(self, serial: str, console_class: Type[Console] = Console) -> None:
+        self._serial = serial
+        self._output_states: Dict[int, bool] = {
+            output.value: False
+            for output in PowerOutputPosition
+        }
+        self._led_states: Dict[int, bool] = {
+            i: False
+            for i in range(2)
+        }
+
+        # Setup console helper
+        self._console = console_class(f"{self.board.__name__}({self._serial})")
+
+    @property
+    def firmware_version(self) -> Optional[str]:
+        """The firmware version reported by the board."""
+        return None  # Console, so no firmware
+
+    @property
+    def serial(self) -> str:
+        """The serial number reported by the board."""
+        return self._serial
+
+    def get_firmware_version(self) -> Optional[str]:
+        """Get the firmware version reported by the board."""
+        return str(self.firmware_version)
+
+    def get_power_output_enabled(self, identifier: int) -> bool:
+        """Get whether a power output is enabled."""
+        try:
+            return self._output_states[identifier]
+        except KeyError:
+            raise ValueError(f"Invalid power output identifier {identifier!r}; "
+                             f"valid identifiers are "
+                             f"{self._output_states.keys()}") from None
+
+    def set_power_output_enabled(
+        self, identifier: int, enabled: bool,
+    ) -> None:
+        """Set whether a power output is enabled."""
+        try:
+            self._console.info(f"Setting output {identifier} to {enabled}")
+            self._output_states[identifier] = enabled
+        except KeyError:
+            raise ValueError(f"Invalid power output identifier {identifier!r}; "
+                             f"valid identifiers are "
+                             f"{self._output_states.keys()}") from None
+
+    def get_power_output_current(self, identifier: int) -> float:
+        """Get the current being drawn on a power output, in amperes."""
+        if identifier in self._output_states:
+
+            return self._console.read(f"Power Output current for {identifier}")
```

Comment:
- I think that this would be better wording given that `identifier` is an integer rather than an output name.
- I think we should make it clear to the user what they're expected to type in (i.e. type or units of value).
- This should be returning a `float` not a `str`.

```suggestion
            return self._console.read(f"Current for power output {identifier} [amps]", float)
```